### PR TITLE
Initial pass added Space

### PR DESCRIPTION
**Pull request by greatbacon**
This PR adds a Space model/interface; defines a table for it & some initial data in the seed file; and has a simple query to read from the DB for a space given an id (testable from the browser url `localhost:3000/api/v1/spaces/2`

### CODE REVIEW

File: src/db/Database.ts [hunk 1]
```diff
@@ -115,5 +116,24 @@ export class Database {
 				return {ok: true, value: data};
 			},
 		},
+		spaces: {
+			findById: async (
+				spaceId: string,
+			): Promise<Result<{value: Space[]}, {type: 'noSpaceFound'; reason: string}>> => {
+				console.log(`[db] preparring to query for space id: ${spaceId}`);
+				const data = await this.sql<Space[]>`
+				select spaceId, url, mediaType, content from space where spaceId = ${spaceId}
```
Comment (ryanatkn) [inline src/db/Database.ts:125]: `space` -> `spaces`

File: src/spaces/space.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface Space {
+	spaceId?: number;
```
Comment (ryanatkn) [inline src/spaces/space.ts:2]: We're running afoul of Postgres' case insensitivity here. We need to either go lowercase in the database or quote everything. I'm partial to underscores for property names, but things like the `mediaType` below are ActivityStreams vocab - which is awesome - but I'm wondering what the best long term convention is.

Comment (greatbacon) [reply src/spaces/space.ts:2]: okay, I wasn't sure if that was a PSQL thing or something overlooked. I'm fine with `snake_case` propagating or maybe creating DB level interfaces with functions to convert them into the broader `camelCase` form?

File: src/db/Database.ts [hunk 3]
```diff
@@ -122,7 +122,7 @@ export class Database {
 			): Promise<Result<{value: Space[]}, {type: 'noSpaceFound'; reason: string}>> => {
 				console.log(`[db] preparring to query for space id: ${spaceId}`);
 				const data = await this.sql<Space[]>`
-				select spaceId, url, mediaType, content from space where spaceId = ${spaceId}
+				select spaceId, url, mediaType, content from spaces where spaceId = ${spaceId}
```
Comment (ryanatkn) [inline src/db/Database.ts:125]: i leave a file unsaved about every week, and it is never not baffling

File: src/spaces/space.ts (hunk 2 above)
Comment (ryanatkn) [reply src/spaces/space.ts:2]: I like that, we can build views with `camelCase`, and we'd probably use views anyway.

File: src/db/seed.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export const seed = async (db: Database): Promise<void> => {
 
 	const createSpacesTableResult = await sql`
 	create table if not exists spaces (
-		spaceId serial primary key,
+		space_id serial primary key,
 		url text,
 		mediaType text,
```
Comment (ryanatkn) [inline src/db/seed.ts:49]: we could call this `mime` for now, or `media_type` if you prefer

Comment (greatbacon) [reply src/db/seed.ts:49]: oh yep, the other 🐍 convert

I'll stick with `media_type` since that's the AS defined term

**Comment by greatbacon** [comment]
@ryanatkn are there any outstanding notes on this PR, or am I good to merge?

**Comment by ryanatkn** [comment]
My bad, I was looking specifically for that media_type commit and I must've gotten tangled with my notifications and missed it.

**Review by ryanatkn — APPROVED** [review]

**Comment by greatbacon** [comment]
@ryanatkn no worries, there was some wonkiness in the change getting pushed yesterday, so I'm not surprised it slipped under the radar. Thanks!

**Pull request merged**